### DISCUSSION TITLE
Does clip support Chinese characters in PowerShell?

### DISCUSSION THREAD

**Discussion by fgsoap** [nushell/nushell#8117] [Q&A] [answered]
### Question

tried in both PowerShell 5&&7, seems not working

`C:\〉echo "你好" | clip  `

got "Σ╜áσÑ╜" after pasting to text editor

**Comment by rgwood** [comment]
Hi, just to confirm: are you running Nushell or PowerShell?

If Nushell, which version? Please run `version | transpose key value | to md --pretty` and paste the output here.

**Reply by fgsoap** [reply]
| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.75.0                                   |
| branch             |                                          |
| commit_hash        | 17a265b197ad6bb77db5a07942a9f160ec260eb0 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| pkg_version        | 0.75.0                                   |
| build_time         | 2023-01-31 20:39:29 +00:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |

**Reply by fgsoap** [reply]
running Nushell in PowerShell via `nu`

**Reply by rgwood** [reply]
Are you using Windows Terminal?

**Reply by fgsoap** [reply]
Yes, it's Windows Terminal.
![image](https://user-images.githubusercontent.com/2555074/219910284-c6c7d943-d341-42e1-bf04-eeb30c3c69fb.png)

**Reply by fgsoap** [reply]
same issue on `cmd.exe`
![image](https://user-images.githubusercontent.com/2555074/219911037-c7e1bba1-8ad3-4106-9aba-a3364e77f1d2.png)

**Reply by fgsoap** [reply]
issue seems happening on non-ASCII characters

**Comment by WindSoilder** [comment, marked as answer]
Just curious, where is clip command comes from?

**Reply by rgwood** [reply]
`clip` ships with Windows: https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/clip
```
〉which clip
╭───┬──────┬──────────────────────────────┬──────────╮
│ # │ arg  │             path             │ built-in │
├───┼──────┼──────────────────────────────┼──────────┤
│ 0 │ clip │ C:\windows\system32\clip.exe │ false    │
╰───┴──────┴──────────────────────────────┴──────────╯
```

**Reply by fgsoap** [reply]
@rgwood Looks the issue is caused by `clip` itself, tried it without Nushell, same issue.

**Reply by fgsoap** [reply]
looks working after this

`cmd.exe`
![image](https://user-images.githubusercontent.com/2555074/219917550-86b89d31-5a0a-42a2-ad98-5a716b958100.png)

`PowerShell`
This also needs to enable with `$OutputEncoding = [console]::InputEncoding = [console]::OutputEncoding = New-Object System.Text.UTF8Encoding`

Ref: https://stackoverflow.com/questions/57131654/using-utf-8-encoding-chcp-65001-in-command-prompt-windows-powershell-window
